lib/msal-browser/src/broker/msg/resp/BrokerAuthResponse.ts

```diff
@@ -29,27 +32,55 @@ export class BrokerAuthResponse extends BrokerMessage {
             message.data.interactionType &&
             (message.data.result || message.data.error)) {
 
-            // TODO: verify version compat
-
-            return new BrokerAuthResponse(message.data.interactionType, message.data.result, message.data.error);
+            return new BrokerAuthResponse(message.data.interactionType, message.data.result, BrokerAuthResponse.detectError(message));
         }
 
         return null;
     }
 
+    static detectError(message: MessageEvent): AuthError {
```

Comment:
I think we discussed moving some of this logic to either its own file in /error or into one of the existing error files. Could we do that in this PR?

Reply:
I've done that in a separate PR. There are a lot of moving parts and changing it here will end up causing issues down the line.